src/utils/utils.ts

```diff
@@ -62,7 +64,93 @@ function standardErrorReplacer(key: string, value: any) {
   return value;
 }
 
-function outputFormatter(msg: OutputObject): string | Uint8Array {
+async function* arrayToAsyncIterable<T>(array: T[]): AsyncIterable<T> {
+  for (const item of array) {
+    yield item;
+  }
+}
+
+// Function to handle 'table' type output
+const outputTableFormatter = async (
+  rowStream: TableRow[] | AsyncIterable<TableRow>,
+  options?: TableOptions,
+): Promise<string> => {
+  let output = '';
+  const maxColumnLengths: Record<string, number> = {};
+  let rowCount = 0;
+
+  // Initialize maxColumnLengths with header lengths if headers are provided
+  if (options?.headers) {
+    for (const header of options.headers) {
+      maxColumnLengths[header] = header.length;
+    }
+  }
+
+  let iterableStream = Array.isArray(rowStream)
+    ? arrayToAsyncIterable(rowStream)
+    : rowStream;
+
+  const updateMaxColumnLengths = (row: TableRow) => {
+    for (const [key, value] of Object.entries(row)) {
+      const cellValue =
+        value === null || value === '' || value === undefined ? 'N/A' : value;
```

Comment:
You can simplify this. If a value is `null` or `undefined` you can use `??` to use a default.

`value ?? 'N/A`, so only `value == ''` needs to be checked. The whole thing can be compressed to `value == '' ? 'N/A' : value ?? 'N/A'`

Reply:
Alright, thanks, that looks much cleaner.

Reply:
I changed this to
```ts
const cellValue = row[key] == null || row[key].length === 0 ? 'N/A' : row[key];
```
so it's a bit more readable